core/src/main/java/com/graphhopper/storage/CHGraphImpl.java

```diff
@@ -853,7 +853,7 @@ int weightToWeightFlags(long edgePointer, double weight) {
             if (weight > MAX_WEIGHT)
                 weightInt = MAX_WEIGHT_31;
             else
-                weightInt = ((int) (weight * WEIGHT_FACTOR)) << 2;
+                weightInt = Math.round((float) (weight * WEIGHT_FACTOR)) << 2;
```

Comment:
In terms of precision wouldn't it be better to do `(int)Math.round(x)`? (If not, then we would need to do this in the EncodedValue classes like here)

Reply:
Oh, no idea tbh! Not sure if its better to cast to float and then round to int or round to long and cast to int. I changed it to the way you proposed.